src/Tickets/Commerce/Order_Modifiers/Custom_Tables/Abstract_Custom_Table.php

```diff
@@ -56,4 +56,26 @@ protected function check_and_add_index( array $results, string $index_name, stri
 		$results[ "{$table_name}.{$columns}" ] = $message;
 		return $results;
 	}
+
+	/**
+	 * Empties the custom table in a way that is not causing an implicit commit.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool Whether it was emptied or not.
+	 */
+	public function truncate(): bool {
+		if ( ! $this->exists() ) {
+			// There is really nothing to do.
+			return true;
+		}
+
+		$this_table = static::table_name( true );
+
+		DB::query( 'SET foreign_key_checks = 0' );
+		$result = DB::query( "DELETE FROM {$this_table}" );
```

Comment:
We dont want to truncate because truncate causes an implicit commit in the DB. We have the `empty` method which uses TRUNCATE. This method pretty much does the same thing other than resetting the AUTO_INCREMENT which we dont care about it now.